src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -0,0 +1,488 @@
+                    }
+                }
+            }
+        }
+
+        public bool SayHelloToBroker(Activity callerActivity)
+        {
+            try
+            {
+                IAccountManagerFuture result = null;
+                // Callback is not passed since it is making a blocking call to get
+                // intent. Activity needs to be launched from calling app
+                // to get the calling app's metadata if needed at BrokerActivity.
+
+                Bundle HelloBundle = new Bundle();
+                HelloBundle.PutString(BrokerConstants.ClientAdvertisedMaximumBPVersionKey, BrokerConstants.BrokerProtocalVersionCode);
+                HelloBundle.PutString(BrokerConstants.ClientConfiguredMinimumBPVersionKey, "2.0");
+                HelloBundle.PutString(BrokerConstants.BrokerAccountManagerOperationKey, "HELLO");
+
+                result = _androidAccountManager.AddAccount(BrokerConstants.BrokerAccountType,
+                    BrokerConstants.AuthtokenType, null, HelloBundle, null,
+                    null, GetPreferredLooper(callerActivity));
+
+                if (result == null)
+                {
+                    _logger.Info("Android account manager AddAccount didn't return any results. ");
+                }
+
+                Bundle bundleResult = (Bundle)result?.Result;
+
+                var bpKey = bundleResult.GetString(BrokerConstants.NegotiatedBPVersionKey);
+                var test = bundleResult.GetString(BrokerConstants.CommonProtocolVersion);
+                if (!String.IsNullOrEmpty(bpKey))
+                {
+                    _logger.Info("Using broker protocol version: " + bpKey);
+                    return true;
+                }
+                else
+                {
+                    throw new MsalException("Could not negotiate protocol version with broker");
+                }
+            }
+            catch
+            {
+                _logger.Error("Error when trying to acquire intent for broker authentication.");
+                throw;
+            }
+        }
+
+        private Handler GetPreferredLooper(Activity callerActivity)
+        {
+            var myLooper = Looper.MyLooper();
+            if (myLooper != null && callerActivity.MainLooper != myLooper)
+            {
+                _logger.Info("myLooper returned. Calling thread is associated with a Looper: " + myLooper.ToString());
+                return new Handler(myLooper);
+            }
+            else
+            {
+                _logger.Info("Looper.MainLooper returned: " + Looper.MainLooper.ToString());
+                return new Handler(Looper.MainLooper);
+            }
+        }
+
+        private void ValidateBrokerRedirectURI(IDictionary<string, string> brokerPayload)
+        {
+            //During the silent broker flow, the redirect URI will be null.
+            if (string.IsNullOrEmpty(GetValueFromBrokerPayload(brokerPayload, BrokerParameter.RedirectUri)))
```

Comment:
qns: is there a way to just check if this is the silent flow, versus checking if the redirecturi is null? it could maybe be null in another case because of a bug that's not related to the silent flow.

Reply:
This is no longer needed as both the interactive and silent calls will require the redirect URI.